File: packages/python/prompt_sanitizer/modes.py
from enum import Enum


class Mode(str, Enum):
    """Detection mode controlling the engine layers that run."""

    FAST = "fast"
    """Regex + secrets only. Sub-millisecond. Zero ML dependencies.
    Catches: email, phone, SSN, credit card, IBAN, IP, crypto, MAC,
    API keys, JWTs, AWS keys, DB connection strings, private keys."""

    SMART = "smart"
    """Regex + secrets + NER (Piiranha mDeBERTa-v3). ~50-200ms on CPU.
    Additionally catches: names, organisations, context-dependent PII.
    Requires: pip install ai-prompt-sanitizer[nlp]"""

    FULL = "full"
    """Everything in SMART plus synthetic replacements and audit logging."""

File: packages/ruby/lib/prompt_sanitizer/modes.rb
# frozen_string_literal: true

module PromptSanitizer
  module Mode
    # Regex + secrets only. Sub-millisecond. Zero dependencies.
    # Catches: email, phone, SSN, credit card, IBAN, IP, crypto, MAC, URL,
    # API keys, JWTs, bearer tokens, AWS keys, DB connection strings,
    # private keys, passport, driving licence, date-of-birth.
    FAST = :fast

    # FAST + NER (informers/distilbert or mitie). ~25–50 ms on CPU.
    # Additionally catches: names, organisations, locations, misc entities.
    # Requires: gem "informers"  (or gem "mitie")
    SMART = :smart

    # Everything in SMART plus synthetic replacements and audit logging.
    # Requires: gem "informers", gem "faker"
    FULL = :full

    ALL = [FAST, SMART, FULL].freeze

    def self.valid?(mode)
      ALL.include?(mode)
    end
  end
end
